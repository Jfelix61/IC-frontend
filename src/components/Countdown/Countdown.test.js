import React from "react";
import ReactDOM from "react-dom";
import {
  render,
  fireEvent,
  getByTestId,
  rerender,
  cleanup
} from "react-testing-library";
import { act } from "react-dom/test-utils";

import Countdown from "./Countdown";

afterEach(cleanup);

describe("Countdown", () => {
  it("renders without crashing", () => {
    const { asFragment } = render(<Countdown initialTimerTime={1000} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays 5 seconds", () => {
    const { container } = render(<Countdown initialTimerTime={5000} />);
    const timerValue = getByTestId(container, "timerValue");
    expect(timerValue.textContent).toBe("00:00:05");
  });

  it("ticks down timer", async () => {
    jest.useFakeTimers();

    const { container } = render(<Countdown initialTimerTime={5000} />);
    let timerValue = await getByTestId(container, "timerValue");
    expect(timerValue.textContent).toBe("00:00:05");

    act(() => {
      jest.advanceTimersByTime(1000);
    });
    timerValue = await getByTestId(container, "timerValue");
    expect(timerValue.textContent).toBe("00:00:04");
  });
});
