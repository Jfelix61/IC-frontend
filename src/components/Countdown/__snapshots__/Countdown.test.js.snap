// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown renders without crashing 1`] = `
<DocumentFragment>
  <h1
    class="countdown "
    data-testid="timerValue"
  >
    00:00:01
  </h1>
</DocumentFragment>
`;
